src/main/groovy/de/dkfz/roddy/bam2fastq/BamToFastqWorkflow.groovy

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2018 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/TheRoddyWMS/BamToFastqPlugin/blob/master/LICENSE.txt).
+ */
+package de.dkfz.roddy.bam2fastq
+
+import de.dkfz.roddy.Roddy
+import de.dkfz.roddy.config.ConfigurationError
+import de.dkfz.roddy.core.DataSet
+import de.dkfz.roddy.core.ExecutionContext
+import de.dkfz.roddy.core.ExecutionContextError
+import de.dkfz.roddy.core.Workflow
+import de.dkfz.roddy.execution.io.ExecutionService
+import de.dkfz.roddy.knowledge.files.BaseFile
+import de.dkfz.roddy.knowledge.files.FileGroup
+import de.dkfz.roddy.tools.LoggerWrapper
+
+/*
+ * Copyright (c) 2018 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+import groovy.transform.CompileStatic
+
+import java.nio.file.Files
+
+@CompileStatic
+class BamToFastqWorkflow extends Workflow {
+
+    public static final LoggerWrapper logger = LoggerWrapper.getLogger("BamToFastqWorkflow")
+
+    private synchronized static Map<DataSet,List<BaseFile>> _bamFilesPerDataset = [:]
+
+    private synchronized static Map<DataSet, Config> _config = [:]
+
+    public static final String TOOL_BAM_LIST_READ_GROUPS = "bamListReadGroups"
+    public static final String TOOL_BAM2FASTQ = "bam2fastq"
+    public static final String TOOL_SORT_FASTQ_SINGLE = "sortFastqSingle"
+    public static final String TOOL_SORT_FASTQ_PAIR = "sortFastqPair"
+    public static final String TOOL_CLEANUP = "cleanup"
+
+    List<BaseFile> getBamFiles() {
+        _bamFilesPerDataset.get(context.dataSet, [])
+    }
+
+    Config getConfig() {
+        _config.get(context.dataSet, null)
+    }
+
+    // Remove sample, sequence protocol etc. from filename patterns, etc. -- implement this stuff later
+    void determineBamFiles() {
+        List<BaseFile> bamFiles = []
```

Comment:
Would this method not better be part of the config object? OR a custom runtime service (which would make it more complex)?